src/Control/Monad/Error/Hoist.hs

```diff
@@ -190,3 +169,19 @@ m <!?> e = do
 
 infixl 8 <!?>
 {-# INLINE (<!?>) #-}
+
+-- | A class for plucking an error @e@ out of a partiality type @t@.
+class PluckError e t m | t -> e where
```

Comment:
How about this?

```
class Monad m => PluckError e t m | t -> e where
  inspectError :: t a -> m (Either e a)
  inspectError a = foldError a (pure . Left) (pure . Right)
  foldError :: t a -> (e -> m x) -> (a -> m x) -> m x
  foldError a f g = inspectError a >>= either f g
  {-# MINIMAL inspectError | foldError #-}
```

Reply:
It does make it require `Monad m`, which means you couldn't (hypothetically) write an instance which plucks into a `Validation`. You can force it with `DefaultSignatures` but you end up not being able to take advantage of those defaults if you want the most general type for the instances:

```haskell
class PluckError e t m | t -> e where
  pluckError :: t a -> m (Either e a)
  default pluckError :: Applicative m => t a -> m (Either e a)
  pluckError = foldError (pure . Left) (pure . Right)

  foldError :: (e -> m r) -> (a -> m r) -> t a -> m r
  default foldError :: Monad m => (e -> m r) -> (a -> m r) -> t a -> m r
  foldError f g = pluckError >=> either f g

  {-# MINIMAL pluckError | foldError #-}

instance (Applicative m, e ~ ()) => PluckError e Maybe m where
  pluckError = pure . maybe (Left ()) Right
  foldError f = maybe (f ())

instance Applicative m => PluckError e (Either e) m where
  pluckError = pure
  foldError = either

instance Monad m => PluckError e (ExceptT e m) m where
  pluckError = runExceptT
  foldError f g = runExceptT >=> either f g

instance Applicative m => PluckError e (Except e) m where
  pluckError = pure . runExcept
  foldError f g = either f g . runExcept
```